### PR TITLE
Inline member inits

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: libsolidity/AST.h [hunk 1]
```diff
@@ -447,6 +450,9 @@ class VariableDeclaration: public Declaration
 	void setType(std::shared_ptr<Type const> const& _type) { m_type = _type; }
 
 	virtual bool isLValue() const override;
+
+	/// Checks that all parameters have allowed types and calls checkTypeRequirements on the body.
```
Comment (chriseth) [inline libsolidity/AST.h]: The docstring does not really fit here.

File: libsolidity/ASTPrinter.cpp [hunk 2]
```diff
@@ -225,7 +225,7 @@ bool ASTPrinter::visit(Return const& _node)
 	return goDeeper();
 }
 
-bool ASTPrinter::visit(VariableDefinition const& _node)
+bool ASTPrinter::visit(VariableDeclarationStatement const& _node)
 {
 	writeLine("VariableDefinition");
```
Comment (chriseth) [inline libsolidity/ASTPrinter.cpp]: Please change those, too.

File: libsolidity/AST_accept.h [hunk 3]
```diff
@@ -475,24 +475,24 @@ void ExpressionStatement::accept(ASTConstVisitor& _visitor) const
...
 
-void VariableDefinition::accept(ASTVisitor& _visitor)
+void VariableDeclarationStatement::accept(ASTVisitor& _visitor)
 {
 	if (_visitor.visit(*this))
 	{
 		m_variable->accept(_visitor);
-		if (m_value)
-			m_value->accept(_visitor);
+		if (m_variable->getValue())
```
Comment (chriseth) [inline libsolidity/AST_accept.h]: Should not be here.

File: libsolidity/AST_accept.h [hunk 4]
```diff
@@ -475,24 +475,24 @@ void ExpressionStatement::accept(ASTConstVisitor& _visitor) const
...
 
-void VariableDefinition::accept(ASTConstVisitor& _visitor) const
+void VariableDeclarationStatement::accept(ASTConstVisitor& _visitor) const
 {
 	if (_visitor.visit(*this))
 	{
 		m_variable->accept(_visitor);
-		if (m_value)
-			m_value->accept(_visitor);
+		if (m_variable->getValue())
```
Comment (chriseth) [inline libsolidity/AST_accept.h]: Should also be removed.

File: libsolidity/Compiler.cpp [hunk 5]
```diff
@@ -247,6 +249,16 @@ void Compiler::registerStateVariables(ContractDefinition const& _contract)
 			m_context.addStateVariable(*variable);
 }
 
+void Compiler::initializeStateVariables(ContractDefinition const& _contract)
+{
+	for(ASTPointer<VariableDeclaration> const& variable: _contract.getStateVariables())
```
Comment (chriseth) [inline libsolidity/Compiler.cpp]: Space missing after `for`

File: libsolidity/Compiler.cpp [hunk 6]
```diff
@@ -247,6 +249,16 @@ void Compiler::registerStateVariables(ContractDefinition const& _contract)
...
+		if (variable->getValue())
+		{
+			compileExpression(*(variable->getValue()), (variable->getValue())->getType());
```
Comment (chriseth) [inline libsolidity/Compiler.cpp]: If you invoke the expression compiler in the next line anyway, why not move this line into `appendStateVariableInitialization`?

File: libsolidity/ExpressionCompiler.cpp [hunk 7]
```diff
@@ -56,6 +56,19 @@ void ExpressionCompiler::appendStateVariableAccessor(CompilerContext& _context,
...
 
+void ExpressionCompiler::appendStateVariableInitialization(CompilerContext& _context, VariableDeclaration const& _varDecl, bool _optimize)
+{
+	ExpressionCompiler compiler(_context, _optimize);
+	compiler.appendStateVariableInitialization(_varDecl);
+}
+
+void ExpressionCompiler::appendStateVariableInitialization(VariableDeclaration const& _varDecl)
+{
+	m_currentLValue.fromStateVariable(_varDecl);
```
Comment (chriseth) [inline libsolidity/ExpressionCompiler.cpp]: No need to use `m_currentLValue`, you can just use a local instance of `LValue`, and then you do not need to call `reset`.

File: libsolidity/ExpressionCompiler.cpp [hunk 8]
```diff
@@ -1018,12 +1030,24 @@ void ExpressionCompiler::LValue::fromIdentifier(Identifier const& _identifier, D
 		m_dataType = _identifier.getType();
 		solAssert(m_dataType->getStorageSize() <= numeric_limits<unsigned>::max(),
 				  "The storage size of " + m_dataType->toString() + " should fit in an unsigned");
-		m_size = unsigned(m_dataType->getStorageSize());	}
```
Comment (chriseth) [inline libsolidity/ExpressionCompiler.cpp]: Please call `fromStateVariable` here.

Comment (LianaHus) [inline libsolidity/ExpressionCompiler.cpp]: and how would you call it? fromDeclaration?

Comment (chriseth) [inline libsolidity/ExpressionCompiler.cpp]: Yes, that sounds good.

File: libsolidity/ExpressionCompiler.cpp [hunk 9]
```diff
@@ -56,6 +56,23 @@ void ExpressionCompiler::appendStateVariableAccessor(CompilerContext& _context,
...
+	if (_varDecl.getValue()->getType())
+		appendTypeConversion(_context, *(_varDecl.getValue())->getType(), *(_varDecl.getValue())->getType());
+
+	ExpressionCompiler compiler(_context, _optimize);
+	compiler.appendStateVariableInitialization(_varDecl);
+}
+
+void ExpressionCompiler::appendStateVariableInitialization(VariableDeclaration const& _varDecl)
+{
+	LValue lvalue = LValue(m_context);
```
Comment (gavofyork) [inline libsolidity/ExpressionCompiler.cpp]: perhaps a more descriptive name than simply a repetition of the typename? `var` or `variable` seem reasonable.

**Pull request merged**